tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_files/invalid_use_of_overload_acceptable.py

```diff
@@ -0,0 +1,31 @@
+# Test file for InvalidUseOfOverload checker
+
+from typing import Awaitable, overload, Union
+
+
+@overload
+def double(a: str)  -> Awaitable[int]:
+    ...
+
+@overload
+def double(a: int) -> Awaitable[int]:
+    ...
+
+async def double(a: Union[str, int]) -> int:
+    if isinstance(a, str):
```

Comment:
shouldn't this fail here?

Reply:
My understanding is that because the overloaded functions have an Awaitable[int] as a return type this makes them an async function too. If this isn't the case it does simplify the implementation.

Reply:
Ah sorry I did not see the awaitable :)